driven/storage/adapter.go

```diff
@@ -2113,6 +2113,33 @@ func (sa *Adapter) ReportPostAsAbuse(clientID string, userID string, group *mode
 	return err
 }
 
+// ReactToPost React to a post
+func (sa *Adapter) ReactToPost(userID string, postID string, reaction string, on bool) error {
+	filter := bson.D{primitive.E{Key: "_id", Value: postID}}
+	update := bson.D{
+		primitive.E{Key: "$set", Value: bson.D{
+			primitive.E{Key: "date_updated", Value: time.Now()},
+		}},
+	}
+	updateOperation := "$pull"
+	if on {
+		updateOperation = "$push"
+	}
+	update = append(update, primitive.E{Key: updateOperation, Value: bson.D{
+		primitive.E{Key: "reactions." + reaction, Value: userID},
+	}})
+
+	res, err := sa.db.posts.UpdateOne(filter, update, nil)
```

Comment:
When a new `Post` is created, the reactions map is null by default which will mean that the `$push` operation will fail. We either need to update this mongo operation (I believe changing to a subdoc instead of the dot notation will work), or ensure that the `reactions` map is never null when stored.

Reply:
Addressed by #249